src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -5226,22 +5790,25 @@ void DynamicIntConvSurfaceClassification(EnergyPlusData &state, int const SurfNu
     using Psychrometrics::PsyWFnTdpPb;
 
     // SUBROUTINE PARAMETER DEFINITIONS:
-    Real64 const g(9.81);                     // gravity constant (m/s**2)
-    Real64 const v(15.89e-6);                 // kinematic viscosity (m**2/s) for air at 300 K
-    Real64 const ActiveDelTempThreshold(1.5); // deg C, temperature difference for surfaces to be considered "active"
+    Real64 constexpr g(9.81);                     // gravity constant (m/s**2)
+    Real64 constexpr v(15.89e-6);                 // kinematic viscosity (m**2/s) for air at 300 K
+    Real64 constexpr ActiveDelTempThreshold(1.5); // deg C, temperature difference for surfaces to be considered "active"
 
     // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
     int ZoneNum(0);
     int PriorityEquipOn(0);
-    Array1D_int HeatingPriorityStack({0, 10}, 0);
-    Array1D_int CoolingPriorityStack({0, 10}, 0);
-    Array1D_int FlowRegimeStack({0, 10}, 0);
+    constexpr int MaxZoneEquipmentOn{11};
+    constexpr int MaxZoneEquipmentIdx{MaxZoneEquipmentOn - 1};
+    std::array<int, MaxZoneEquipmentOn> HeatingPriorityStack{};
+    std::array<int, MaxZoneEquipmentOn> CoolingPriorityStack{};
+    std::array<ConvectionConstants::InConvFlowRegime, MaxZoneEquipmentOn> FlowRegimeStack{};
+    FlowRegimeStack.fill(ConvectionConstants::InConvFlowRegime::Invalid);
```

Comment:
Replaced the magic number 10 with a constexpr int. See here for futher discussion: https://github.com/NREL/EnergyPlus/pull/8862#discussion_r663145869

Reply:
Choosing 10 was lazy, coulda shoulda dynamically checked the model for the highest number of zone equipment attached to a zone. 10 is plenty but overkill.